nautobot/extras/migrations/0094_ensure_note_timestamps_are_unique.py

```diff
@@ -0,0 +1,40 @@
+# Generated by Django 3.2.20 on 2023-07-18 14:50
+
+import random
+
+from datetime import timedelta
+from django.db import migrations
+from django.db import models
+
+
+def ensure_note_created_timestamps_are_unique(apps, schema_editor):
+    Note = apps.get_model("extras", "Note")
+    natural_key_fields = ["assigned_object_type", "assigned_object_id", "user_name", "created"]
+
+    # We append some random milliseconds of time to avoid duplicate timestamps for Note objects' created field.
+    duplicate_records = (
+        Note.objects.values(*natural_key_fields)
+        .order_by()
+        .annotate(count=models.Count(natural_key_fields))
+        .filter(count__gt=1)
+    )
+    for duplicate_record in duplicate_records:
+        duplicate_notes = Note.objects.filter(
+            assigned_object_type=duplicate_record.get("assigned_object_type"),
+            assigned_object_id=duplicate_record.get("assigned_object_id"),
+            user_name=duplicate_record.get("user_name"),
+        )
```

Comment:
I had to explicitly filter on each field here instead of doing `**duplicate_record` because ** expression fails on `mysql` for some reason.